File: NeetCode/10-Heap-Priority-Queue/KthLargestElementinaStream.cpp
#include <vector>
#include <queue>
using namespace std;
class KthLargest {
private: 
    priority_queue<int, vector<int>, greater<int>> heap;
    int k;
public:
    KthLargest(int k, vector<int>& nums) {
        this->k = k;
        for(int num : nums){
            add(num);
        }
    }
    
    int add(int val) {
        heap.push(val);
        if(heap.size() > k){
            heap.pop();
        }
        return heap.top();
    }
};

File: NeetCode/10-Heap-Priority-Queue/KthLargestElementinaStream.java
import java.util.*;
public class KthLargestElementinaStream {
    PriorityQueue<Integer> heap;
    int k;
    public KthLargestElementinaStream(int k, int[] nums) {
       this.k = k;
       this.heap = new PriorityQueue<>();
       for(int i = 0; i<nums.length; i++){
        add(nums[i]);
       }
    }
    
    public int add(int val) {
        heap.offer(val);
        if(heap.size() > k){
            heap.poll();
        }
        return heap.peek();
    }
}
